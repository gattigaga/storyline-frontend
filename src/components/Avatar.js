import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Image = styled.img`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;
  border-radius: 100%;
`;

const Avatar = ({ className, src, size, alt }) => (
  <Image src={src} size={size} alt={alt} className={className} />
);

Avatar.propTypes = {
  src: PropTypes.string,
  size: PropTypes.number,
  alt: PropTypes.string,
  className: PropTypes.string
};

Avatar.defaultProps = {
  size: 64
};

export default Avatar;
