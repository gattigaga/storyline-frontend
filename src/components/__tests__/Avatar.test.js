import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Avatar from "../Avatar";

describe("Avatar", () => {
  const setup = propOverrides => {
    const props = {
      src: require("../../../assets/images/dummy.png"),
      alt: "Profile",
      ...propOverrides
    };

    const wrapper = shallow(<Avatar {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom size", () => {
    const { wrapper } = setup({ size: 96 });

    expect(wrapper.props().size).toEqual(96);
  });
});
