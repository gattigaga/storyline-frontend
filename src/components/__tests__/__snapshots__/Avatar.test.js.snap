// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should renders default 1`] = `
<styled.img
  alt="Profile"
  size={64}
  src="dummy.png"
/>
`;
